src/ring_buf/ring_buf.rs

```diff
@@ -0,0 +1,99 @@
+use crate::error::BccError;
+use crate::ring_buf::callback::{raw_callback, RingCallback};
+use crate::table::Table;
+use crate::types::MutPointer;
+
+use bcc_sys::bccapi::{
+    bpf_add_ringbuf, bpf_consume_ringbuf, bpf_free_ringbuf, bpf_new_ringbuf, bpf_poll_ringbuf,
+    ring_buffer,
+};
+
+/// A builder type for producing `PerfMap`s
+pub struct RingBufBuilder {
+    table_cb_pairs: Vec<(Table, RingCallback)>,
+}
+
+impl RingBufBuilder {
+    /// Create a new builder for a given ring buffer and callback function
+    pub fn new(table: Table, rcb: RingCallback) -> Self {
+        Self {
+            table_cb_pairs: vec![(table, rcb)],
+        }
+    }
+
+    /// Add a ring buffer and callback function
+    pub fn add(mut self, table: Table, rcb: RingCallback) -> Self {
+        self.table_cb_pairs.push((table, rcb));
+        self
+    }
+
+    /// Try constructing a `RingBuf` from the builder
```

Comment:
Take a look at how we handle raw tracepoints with bcc version dependency. https://github.com/rust-bpf/rust-bcc/blob/master/src/raw_tracepoint/mod.rs#L36